Make experiment

Jdvdb opened the pull request:
Dialog created that is identical to the design in the storyboard (except for the locale field since I wasn't sure the purpose of it).
Added the ability to create experiments with more information in the ExperimentMaker based on inputs used in the Dialog.
Added a small button to MainActivity and implemented code to easily test the Dialog on this screen via the debug console.
Javadocs comments were written and citations were added where appropriate in code.
I have no unit testing since from what I can tell there's no simple way to do it on a dialog.
Added attributes to experiment class to tell you if an experiment is accepting new results and if it records information.
Utilized the strings and color resource section in the creation of the dialog

giancarlopernudisegura reviewed (changes requested):
Could you add some padding to the fab? Something like 16dp should be fine. Also can you add javadoc comments to the constructors including the parameter values? Otherwise everything else looks good.

Jdvdb commented:
Added javadocs for abstract class and all four experiments for their constructors. 
Added some space to the FAB so that it looks a bit nicer
Fixed typo in IllegalException error that was created & refactored all associated code
